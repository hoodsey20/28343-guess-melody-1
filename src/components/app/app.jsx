import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

const App = ({questions, renderScreen, question}) => {
  return (
    <React.Fragment>
      {renderScreen(questions, question)}
    </React.Fragment>
  );
};

App.propTypes = {
  // store state
  question: PropTypes.number.isRequired,

  renderScreen: PropTypes.func.isRequired,
  questions: PropTypes.arrayOf(
      PropTypes.oneOfType(
          [
            PropTypes.shape({
              answers: PropTypes.arrayOf(PropTypes.shape({
                artist: PropTypes.string.isRequired,
                picture: PropTypes.string.isRequired,
              })).isRequired,
              song: PropTypes.shape({
                artist: PropTypes.string.isRequired,
                src: PropTypes.string.isRequired,
              }).isRequired,
              type: PropTypes.oneOf([`artist`]).isRequired,
            }),
            PropTypes.shape({
              answers: PropTypes.arrayOf(PropTypes.shape({
                src: PropTypes.string.isRequired,
                genre: PropTypes.oneOf([`rock`, `jazz`, `pop`]).isRequired,
              })).isRequired,
              genre: PropTypes.oneOf([`rock`, `jazz`, `pop`]).isRequired,
              type: PropTypes.oneOf([`genre`]).isRequired,
            })
          ]
      )
  ).isRequired,
};

export {App};

const mapStateToProps = (state) => ({
  question: state.question,
});

export default connect(mapStateToProps)(App);
