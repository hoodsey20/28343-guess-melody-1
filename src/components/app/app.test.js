import React from 'react';
import renderer from 'react-test-renderer';

import {App} from './app.jsx';

const mock = {
  question: -1,
  questions: [
    {
      type: `genre`,
      genre: `rock`,
      answers: [
        {
          src: ``,
          genre: `rock`,
        },
      ],
    }, {
      type: `artist`,
      song: {
        artist: `Jim Beam`,
        src: ``,
      },
      answers: [
        {
          picture: ``,
          artist: `John Snow`,
        },
      ],
    },
  ]
};

const renderScreenMock = () => <h1>app</h1>;

it(`App is rendered correctly`, () => {
  const {questions, question} = mock;
  const tree = renderer.create(<App
    questions={questions}
    question={question}
    renderScreen={renderScreenMock}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
