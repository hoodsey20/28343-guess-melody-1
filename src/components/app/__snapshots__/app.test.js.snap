// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is rendered correctly 1`] = `
<h1>
  app
</h1>
`;
